lib/Models/CatalogItem.js

```diff
@@ -984,7 +999,14 @@ function isEnabledChanged(catalogItem) {
             catalogItem._loadForEnablePromise = resolvedOrRejected ? undefined : loadPromise;
         }
 
-        catalogItem.isShown = true;
+        if (catalogItem.isShown) {
+            // If item already had the isShown flag but wasn't enabled, it wasn't actually shown, so call isShownChanged
+            // to update the view.
```

Comment:
Is this still need after the changes to make the disclaimer sort of hook into the load process?  I can't think of why it would be, but maybe I've missed something.